src/vs/workbench/contrib/debug/browser/variablesView.ts

```diff
@@ -123,9 +123,10 @@ export class VariablesView extends ViewPane {
 		container.classList.add('debug-variables');
 		const treeContainer = renderViewTree(container);
 		const linkDetector = this.instantiationService.createInstance(LinkDetector);
+		const displayType: boolean = this.configurationService.getValue('variableDisplayType.variableView');
 		this.tree = <WorkbenchAsyncDataTree<IStackFrame | null, IExpression | IScope, FuzzyScore>>this.instantiationService.createInstance(WorkbenchAsyncDataTree, 'VariablesView', treeContainer, new VariablesDelegate(),
 			[
-				this.instantiationService.createInstance(VariablesRenderer, linkDetector),
+				this.instantiationService.createInstance(VariablesRenderer, linkDetector, displayType),
```

Comment:
This doesn't listen to changes, so changing the setting requires an editor restart.

It probably makes sense for the VariablesRenderer to listen to it and automatically re-render all rendered elements when it changes.

Reply:
What would be the best way to get this behaviour?

Reply:
You can listen to configuration changes on the `IConfigurationService`, something like

```js
configurationService.onDidChangeConfiguration(e => {
	if (e.affectsConfiguration('you configuration key')) {
		// rerender
	}
})

Reply:
Hi, we have already done everything, only the rerender is missing. How exactly do we trigger a rerender?
We tried `this.tree.rerender()`, `this.tree.updateChildren()` and we also tried calling the function itself again like `this.renderBody(container)`. We tried to console.log something and the listener is really working, when we change configuration it prints to the console, we just don't know how to trigger the rerender.

Reply:
I would implement it in the renderer to cause it to call a function to render its element again when the event happens. For example in the test view I do this:

https://github.com/microsoft/vscode/blob/4a4ae4603279ad4d836c4d2b17f055f6708d59ff/src/vs/workbench/contrib/testing/browser/testingExplorerView.ts#L1450

Reply:
I don't think I quite understood you. 

I don't know why something like this doesn't work? Every time the configuration changes, the tree is re-rendered with the new type display setting.

src/vs/workbench/contrib/debug/browser/variablesView.ts
![image](https://github.com/microsoft/vscode/assets/25253620/46af8f06-360e-4585-9765-f5001c5d5ed5)


I think what you are suggesting is something like this:

src/vs/workbench/contrib/debug/browser/variablesView.ts
![image](https://github.com/microsoft/vscode/assets/25253620/5ba432ae-f17b-4ee9-b02f-6611dc50b1cb)

but this has several problems, first in IExpression there are no events and even if I added one how can I know when to emit it when I only have access to IConfigurationService in VariablesView class and now I am at VariablesRenderer.

Thanks for the help!

Reply:
Yea, the latter case is what I was thinking. In that case you would listen to the configuration service in the variables renderer, you can inject it through the VariablesRenderer's constructor parameters by adding `@IConfigurationService private configurationService: IConfigurationService`.

Another simpler approach is to always insert the type element, but have it conditionally displayed based on a CSS class applied to the tree, then that can be toggled on the tree / container based on configuration.

Reply:
Thanks for the tip!